crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -272,15 +274,43 @@ async fn download_new_classes(
             .await
             .with_context(|| format!("Downloading class {}", class_hash.0))?;
 
-        tx_event
-            .send(Event::NewContract(class))
-            .await
-            .with_context(|| {
-                format!(
-                    "Sending Event::NewContract for declared class {}",
-                    class_hash.0
-                )
-            })?;
+        match class {
+            DownloadedClass::Cairo(class) => tx_event
+                .send(Event::NewCairoContract(class))
+                .await
+                .with_context(|| {
+                    format!(
+                        "Sending Event::NewCairoContract for declared class {}",
+                        class_hash.0
+                    )
+                })?,
+            DownloadedClass::Sierra(sierra_class, casm_class) => {
+                let compiled_class_hash = state_diff
+                    .declared_classes
+                    .iter()
+                    .find_map(|declared_class| {
+                        if declared_class.class_hash.0 == class_hash.0 {
+                            Some(declared_class.compiled_class_hash)
+                        } else {
+                            None
+                        }
+                    })
+                    .ok_or_else(|| anyhow::anyhow!("Sierra class hash not in declared classes"))?;
```

Comment:
Or maybe `with_context` if the hash is important to know.
```suggestion
                    .context("Sierra class hash not in declared classes")?;
```